lib/ston_api_client.ex

```diff
@@ -1,18 +1,53 @@
 defmodule StonApiClient do
   @moduledoc """
-  Documentation for `StonApiClient`.
+  A client for working with the API of a decentralized automated market maker Ston.fi
   """
 
+  alias StonApiClient.Asset
+
+  @base_url "https://api.ston.fi/v1"
+
   @doc """
-  Hello world.
+  Getting information about all assets
+  """
+  def assets do
+    with {:ok, data} <- make_request("#{@base_url}/assets") |> parse_body,
+         {:ok, raw_assets_list} <- Map.fetch(data, "asset_list") do
+      {
+        :ok,
+        raw_assets_list
+        |> Enum.map(&to_atom_keys/1)
+        |> Enum.map(&struct(Asset, &1))
+      }
+    end
+  end
 
-  ## Examples
+  defp make_request(url) do
+    case HTTPoison.get(url) do
+      {:ok, %HTTPoison.Response{status_code: 200, body: body}} ->
+        {:ok, body}
 
-      iex> StonApiClient.hello()
-      :world
+      {:ok, %HTTPoison.Response{status_code: 404}} ->
+        {:error, reason: :not_found}
 
-  """
-  def hello do
-    :world
+      {:error, %HTTPoison.Error{reason: reason}} ->
+        {:error, reason: reason}
+    end
+  end
+
+  defp parse_body({:ok, body}) do
+    case Jason.decode(body) do
```

Comment:
```suggestion
    case Jason.decode(body, keys: :atoms!) do
```

https://hexdocs.pm/jason/Jason.html#decode/2

Есть нюанс, про который необходимо помнить: атомы не собираются сборщиком мусора, а потому чужие данные прогонять через `String.to_atom/1` теоретически чревато переполнением таблицы атомов. Подробнее: https://hexdocs.pm/elixir/String.html#to_existing_atom/1